std/range/package.d

```diff
@@ -6629,10 +6627,20 @@ if (isForwardRange!RangeOfRanges &&
         }
     }
 
-    // ElementType opIndex(size_t n)
-    // {
-    //     return _input[n].front;
-    // }
+    static if (isRandomAccessRange!RangeOfRanges)
+    {
+        auto ref opIndex(size_t n)
+        {
+            auto r = this.save;
+            r._input = _input.dup;
+            while(n)
+            {
+                r.popFront();
+                n--;
+            }
+            return r.front.array;
```

Comment:
No, this is definitely not what we want. You only want to support indexing if the RangeOfRange's element type supports indexing. Then you "index on demand". All the accesses should be O(1).

Reply:
Forgot the .array there, was testing something.

Reply:
To be clear, that wasn't the reason for my critique, yes the array call was unnecessary, but I don't think the `popFront` loop is necessary either.